packages/next-server/server/router.ts

```diff
@@ -21,8 +26,19 @@ export default class Router {
     this.routes.unshift(route)
   }
 
-  match(req: IncomingMessage, res: ServerResponse, parsedUrl: UrlWithParsedQuery) {
-    if (req.method !== 'GET' && req.method !== 'HEAD') {
+  match(
+    req: IncomingMessage,
+    res: ServerResponse,
+    parsedUrl: UrlWithParsedQuery,
+  ) {
+    const isApi = req.url!.match(/^\/api\//)
+
+    if (
```

Comment:
In my opinion we can drop this `if`, we can just render on `POST` / `PUT` requests.